errbot/backends/irc.py

```diff
@@ -304,6 +304,11 @@ def __init__(self,
         super().__init__([(server, port, password)], nickname, username, reconnection_interval=reconnect_on_disconnect)
 
     def connect(self, *args, **kwargs):
+        # Decode all input to UTF-8, but use a replacement character for
+        # unrecognized byte sequences
+        # (as described at https://pypi.python.org/pypi/irc)
+        self.connection.buffer_class.errors = 'replace'
```

Comment:
@perlop Would you find this sufficient? I realize that your issue #569 suggested that all input that is impossible to decode should be ignored -- this seems like a bit more robust solution.